### PR TITLE
[HW-6] Add installation guide to `README.md` file

### PR DESCRIPTION
**Pull request by AlexLevshenkov**
## List of changes

- **Build `.xcframework` by yourself** section has been transformed to **Installation Guide**
- The instructions on how to install the framework via SPM and by downloading `*.xcframework` file have been added

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,9 +17,40 @@ Library containing various flavours of Last-In-First-Out (LIFO) data structures.
...
+The [Swift Package Manager](https://swift.org/package-manager/) is a tool for automating the distribution of Swift code and is integrated into the `swift` compiler. `LifoContainers` supports SPM from `0.2.0` version.
+
+#### Add dependency to the project
+
+You can add `LifoContainers` to an Xcode project by [adding it as a package dependency](https://developer.apple.com/documentation/xcode/adding-package-dependencies-to-your-app#Add-a-package-dependency):
+1. In Xcode, select "File" → "Swift Packages" → "Add Package Dependency";
+2. Enter "https://github.com/AlexLevshenkov/LifoContainers" into the package repository URL text field;
+
+#### Add dependency to the Swift package
+
```
Comment (Nikitapls) [inline README.md:33]: I would suggest you to add the note about minimum versions to this section and remove it from the original place.
```suggestion

**Warning** `LifoContainers` supports SPM from `0.2.0` version.
```

File: README.md [hunk 2]
```diff
@@ -17,9 +17,40 @@ Library containing various flavours of Last-In-First-Out (LIFO) data structures.
...
+Once you have your Swift package set up, adding `LifoContainers` as a dependency is as easy as adding it to the `dependencies` value of your `Package.swift`:
+
+```swift
+dependencies: [
+    .package(url: "https://github.com/AlexLevshenkov/LifoContainers", from: "<version>")
+]
+```
+
+### XCFrameworks
```
Comment (Nikitapls) [inline README.md:42]: nit:
```suggestion
### XCFramework
```

File: README.md [hunk 3]
```diff
@@ -17,9 +17,40 @@ Library containing various flavours of Last-In-First-Out (LIFO) data structures.
...
+
+Each [release](https://github.com/AlexLevshenkov/LifoContainers/releases) includes `*.xcframework` framework binaries.
```
Comment (Nikitapls) [inline README.md:44]: nit:
```suggestion
Each [release](https://github.com/AlexLevshenkov/LifoContainers/releases) includes `*.xcframework` framework binary.
```

File: README.md [hunk 4]
```diff
@@ -17,9 +17,40 @@ Library containing various flavours of Last-In-First-Out (LIFO) data structures.
...
+
+1. Download `*.xcframework.zip` from assets and unpack it;
+2. Open the Xcode project or workspace.
+3. Go to the target’s General configuration page.
+4. Add the framework target to the **Frameworks, Libraries, and Embedded Content** section by clicking the **Add(+)** button
```
Comment (Nikitapls) [inline README.md:49]: I would add a note that in most cases user should choose the `Do not embed` option.
```suggestion
4. Add the framework target to the **Frameworks, Libraries, and Embedded Content** section by clicking the **Add(+)** button. We recommend you to choose the `Do not embed` option.
```

Comment (Nikitapls) [inline README.md:49]:
```suggestion
4. Add the framework target to the **Frameworks, Libraries, and Embedded Content** section by clicking the **Add(+)** button.
```

**Review by Nikitapls — CHANGES_REQUESTED** [review]
PTAL at unresolved conversations

File: README.md [hunk 5]
```diff
@@ -17,9 +17,40 @@ Library containing various flavours of Last-In-First-Out (LIFO) data structures.
 - Xcode 15.2+
 - Swift 5.9+
 
-## Build `.xcframework` by yourself
+## Installation Guide
 
-You may want to build the binary `.xcframework` file yourself from the source code. To do so you need to run the `build-xcframework.sh` script in a [scripts](./scripts/) folder or you may follow [instructions](https://developer.apple.com/documentation/xcode/creating-a-multi-platform-binary-framework-bundle#Create-archives-for-frameworks-or-libraries) provided by Apple.
+### Swift Package Manager
+
+The [Swift Package Manager](https://swift.org/package-manager/) is a tool for automating the distribution of Swift code and is integrated into the `swift` compiler. `LifoContainers` supports SPM from `0.2.0` version.
```
Comment (auhustsinovich) [inline README.md:24]: nit: I can see only `0.1.0` version of your framework, I think you can add this info after release `0.2.0`. For now it doesn`t make any sense.

File: README.md [hunk 6]
```diff
@@ -17,9 +17,40 @@ Library containing various flavours of Last-In-First-Out (LIFO) data structures.
...
+
+#### Add dependency to the project
+
+You can add `LifoContainers` to an Xcode project by [adding it as a package dependency](https://developer.apple.com/documentation/xcode/adding-package-dependencies-to-your-app#Add-a-package-dependency):
+1. In Xcode, select "File" → "Swift Packages" → "Add Package Dependency";
+2. Enter "https://github.com/AlexLevshenkov/LifoContainers" into the package repository URL text field;
```
Comment (auhustsinovich) [inline README.md:30]: You have already provided a link to documentation where user can find installation guide, don't teach them how to use Xcode 👀 

```suggestion
You can add `LifoContainers` to an Xcode project by [adding it as a package dependency](https://developer.apple.com/documentation/xcode/adding-package-dependencies-to-your-app#Add-a-package-dependency).
```

File: README.md [hunk 7]
```diff
@@ -17,9 +17,40 @@ Library containing various flavours of Last-In-First-Out (LIFO) data structures.
...
+
+#### Add dependency to the Swift package
+
+Once you have your Swift package set up, adding `LifoContainers` as a dependency is as easy as adding it to the `dependencies` value of your `Package.swift`:
```
Comment (auhustsinovich) [inline README.md:34]: I'm a little confused, do I have to create a swift package file or not? If not, make it explicit that it's an option to choose to either install through Xcode or through the file directly.

**Review by auhustsinovich — CHANGES_REQUESTED** [review]
please consider following improvements:

File: README.md (hunk 2 above)
Comment (AlexLevshenkov) [reply README.md]: Section has been renamed

File: README.md (hunk 3 above)
Comment (AlexLevshenkov) [reply README.md]: Your suggestion has been applied

File: README.md (hunk 7 above)
Comment (AlexLevshenkov) [reply README.md]: If you want to use `LifoContainers` in your own custom framework(with SPM support) this is the way how to integrate my framework into your framework.

File: README.md (hunk 6 above)
Comment (AlexLevshenkov) [reply README.md]: I don't 🙂. The documentation is broader, and I added 2 actions exactly for my Swift package. The user can quickly connect the framework via the 2 moves, or go deeper inside the Apple documentation, then search somewhere for the link to the my git repository which is inconvenient.

File: README.md (hunk 4 above)
Comment (AlexLevshenkov) [reply README.md]: The user actually shouldn't choose, it is only his decision how he is going to connect the framework.

Comment (AlexLevshenkov) [reply README.md]: This peace of text has been refactored

Comment (Nikitapls) [reply README.md]: We agreed that this is a nit, I leave this up to you.

**Review by Nikitapls — APPROVED** [review]
LGTM

File: README.md (hunk 4 above)
Comment (AlexLevshenkov) [reply README.md]: I have added some text regarding it

File: README.md (hunk 5 above)
Comment (AlexLevshenkov) [reply README.md:24]: The plan is to create tag `0.2.0` and then the release after this PR is merged to let users see the installation guide when they switch to `0.2.0` tag.

**Review by auhustsinovich — APPROVED** [review]
LGTM

**Comment by AlexLevshenkov** [comment]
The pull request had been reviewed and then approved. All conversations were resolved. Merged

**Pull request merged**